src/utils/index.coffee

```diff
@@ -116,23 +117,68 @@ appendToCacheKey = (cache, key, value) ->
   currentValue = cache.get(key) ? []
   cache.set key, [...currentValue, value]
 
+ensureArray = (val) ->
+  return val if isArray val
+  [val]
+
+NON_ROOT_DIRECTORY_IGNORE_PATTERN_REGEX = ///
+  ^
+  [^/]
+  [^.] *
+  $
+///
+getIsNonRootDirectoryPattern = (ignorePattern) ->
+  NON_ROOT_DIRECTORY_IGNORE_PATTERN_REGEX.test ignorePattern
+
 createDirectoryCache = ({
   directory
   recursive
   extensions
   allowed
   context
   context: {settings}
+  ignore
 }) ->
   cache = new Map()
+  getIgnoreMatcher = (
```

Comment:
Actually achieving the `.gitignore`-style matching proved nontrivial

I ended up using [`picomatch`](https://github.com/micromatch/picomatch) to match each directory/filename against any specified ignore patterns as we're recursively traversing the directory listings

And in order to support the `.gitignore`-style "selectors", I use a couple variants of the matcher with different options/massagings of the ignore patterns. This is a little hairy and there may be a simpler way?